src/main/java/derek/Storage.java

```diff
@@ -0,0 +1,145 @@
+            }
+            Scanner getList = new Scanner(tasks);
+            while (getList.hasNext()) {
+                String task = getList.nextLine();
+                taskList.populateTaskList(task);
+            }
+            this.overrideFile();
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        return taskList;
+
+    }
+
+    /**
+     * Overrides the task data file by clearing its contents.
+     *
+     * @throws IOException if an I/O error occurs during file writing
+     */
+    public void overrideFile() throws IOException {
+        FileWriter writer = new FileWriter(dataFilePathName);
+        writer.write("");
+        writer.close();
+
+    }
+
+    /**
+     * Stores the current tasks from the {@code TaskList} into the data file.
+     * Each task is written to the file in a specific format based on its type (e.g., DeadlineTask, EventTask).
+     */
+    public void storeInFile() {
+        try {
+            for (int i = 0; i < taskList.size(); i++) {
+                Task task = taskList.get(i);
+                String isCompleted = task.isCompleted();
+                if (task instanceof DeadlineTask) {
+                    DeadlineTask deadlineTask = (DeadlineTask) task;
+                    String deadline = deadlineTask.getDeadline();
+                    String taskName = deadlineTask.getName();
+                    writeToFile("D|"
+                            + isCompleted
+                            + "|"
+                            + taskName
+                            + "|"
+                            + deadline
+                            + "\n");
+                } else if (task instanceof EventTask) {
+                    EventTask eventTask = (EventTask) task;
+                    String startTime = eventTask.getStartTime();
+                    String endTime = eventTask.getEndTime();
+                    String taskName = eventTask.getName();
+                    writeToFile("E|"
+                            + isCompleted
+                            + "|"
+                            + taskName
+                            + "|"
+                            + startTime
+                            + "|"
+                            + endTime
+                            + "\n");
+                } else {
+                    String taskName = task.getName();
+                    writeToFile("T|"
+                            + isCompleted
+                            + "|" + taskName
+                            + "\n");
+                }
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Writes a given string to the task data file.
+     *
+     * @param textToAdd the string to be added to the file
+     * @throws IOException if an I/O error occurs during file writing
```

Comment:
Missing punctuation behind @param description and @throws description. I noticed the same issue in several other places too.
```suggestion
     * @param textToAdd the string to be added to the file.
     * @throws IOException if an I/O error occurs during file writing.
```